src/Illuminate/Database/Migrations/Migrator.php

```diff
@@ -241,12 +254,16 @@ protected function runUp($file, $batch, $pretend)
             return $this->pretendToRun($migration, 'up');
         }
 
-        $this->write(Task::class, $name, fn () => $this->runMigration($migration, 'up'));
+        if ($this->shouldSkipMigration($migration)) {
+            $this->write(Task::class, $name, fn () => MigrationResult::Skipped);
```

Comment:
Shouldn't this return here instead of the else?

This will log the migration even though it was never run? This means that the migration won't get run later either, after the "feature" is "enabled" or `Migration::shouldRun()` otherwise returns `true`;

```php

        if ($this->shouldSkipMigration($migration)) {
            $this->write(Task::class, $name, fn () => MigrationResult::Skipped);
            return;
         }
```

Reply:
Ahh I see, I was confused, you wrapped the `Migrator::log()` call inside the else block. In that case I would suggest just returning early.